cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -0,0 +1,380 @@
+        """
+        Return a list of certificate identifiers that are already in use for this course
+        """
+        if not course.certificates or not course.certificates.get('certificates'):
+            return []
+        return [cert['id'] for cert in course.certificates['certificates']]
+
+    @staticmethod
+    def assign_id(course, certificate_data, certificate_id=None):
+        """
+        Assign an identifier to the provided certificate data.
+        If the caller did not provide an identifier, we autogenerate a unique one for them
+        In addition, we check the certificate's signatories and ensure they also have unique ids
+        """
+        used_ids = CertificateManager.get_used_ids(course)
+        if certificate_id:
+            certificate_data['id'] = int(certificate_id)
+        else:
+            certificate_data['id'] = generate_int_id(
+                CERTIFICATE_MINIMUM_ID,
+                MYSQL_MAX_INT,
+                used_ids
+            )
+
+        for index, signatory in enumerate(certificate_data['signatories']):  # pylint: disable=unused-variable
+            if signatory and not signatory.get('id', False):
+                signatory['id'] = generate_int_id(used_ids=used_ids)
+            used_ids.append(signatory['id'])
+
+        return certificate_data
+
+    @staticmethod
+    def to_json(certificate):
+        """
+        Serialize the Certificate object's locally-stored certificate data to a JSON representation
+        We use direct access here for specific keys in order to enforce their presence
+        """
+        # pylint: disable=no-member
+        certificate_data = certificate.get_certificate_data()
+        return {
+            "id": certificate_data['id'],
+            "name": certificate_data['name'],
+            "description": certificate_data['description'],
+            "version": CERTIFICATE_SCHEMA_VERSION,
+            "signatories": certificate_data['signatories']
+        }
+
+    @staticmethod
+    def from_json(course, value):
+        """
+        Deserialize from a JSON representation into a Certificate object.
+        'value' should be either a Certificate instance, or a valid JSON string
+        """
+
+        # Ensure the schema fieldset meets our expectations
+        for key in ("name", "description", "version"):
+            if key not in value:
+                raise CertificateValidationError(_("Certificate dict {0} missing value key '{1}'").format(value, key))
+
+        # Load up the Certificate data
+        certificate_data = CertificateManager.parse(value)
+        CertificateManager.validate(certificate_data)
+        certificate_data = CertificateManager.assign_id(course, certificate_data, certificate_data.get('id', None))
+        certificate = Certificate(course, certificate_data)
+
+        # Return a new Certificate object instance
+        return certificate
+
+    @staticmethod
+    def deserialize_certificate(course, json_string):
```

Comment:
Optional: make `deserialize_certificate` a `@classmethod` instead then can call `cls.from_json` - that way if one (anyone) inherits from `CertificateManager` and overrides from_json, the call to `deserialize_certificate` will do the right thing.